### ISSUE TITLE
NSMallocException Failed to grow buffer

### ISSUE THREAD

**Issue by SudoPlz** [Instabug/Instabug-React-Native#198]
Here's an issue we see in production that makes our app crash.

```
NSMallocException Failed to grow buffer 
    Frameworks/CoreFoundation.framework/CoreFoundation ___exceptionPreprocess
    /usr/lib/libobjc.A.dylib _objc_exception_throw
    Frameworks/Foundation.framework/Foundation __NSOutOfMemoryErrorHandler
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFSafelyReallocate
    Frameworks/Foundation.framework/Foundation __NSMutableDataGrowBytes
    Frameworks/Foundation.framework/Foundation -[NSConcreteMutableData appendBytes:length:]
    Frameworks/CoreFoundation.framework/CoreFoundation _writeBytes
    Frameworks/CoreFoundation.framework/CoreFoundation __appendString
    Frameworks/CoreFoundation.framework/CoreFoundation __appendObject
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFBinaryPlistWriteOrPresize
    Frameworks/Foundation.framework/Foundation -[NSKeyedArchiver finishEncoding]
    Frameworks/Foundation.framework/Foundation +[NSKeyedArchiver archivedDataWithRootObject:]
    Frameworks/InstabugCore.framework/InstabugCore -[IBGCache writeModels:toPath:]
    Frameworks/InstabugCore.framework/InstabugCore -[IBGCache persistAll]
    Frameworks/InstabugCore.framework/InstabugCore -[IBGCacheManager removeCacheModel:]
    Frameworks/InstabugCore.framework/InstabugCore -[IBGSession remove]
    Frameworks/InstabugCore.framework/InstabugCore -[IBGSessionRequest serverExecutionDone:statusCode:error:response:]
    Frameworks/InstabugCore.framework/InstabugCore __57-[IBGServerManager URLSession:task:didCompleteWithError:]_block_invoke_2
    /usr/lib/system/libdispatch.dylib __dispatch_call_block_and_release
    /usr/lib/system/libdispatch.dylib __dispatch_client_callout
    /usr/lib/system/libdispatch.dylib __dispatch_main_queue_callback_4CF$VARIANT$mp
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
    Frameworks/CoreFoundation.framework/CoreFoundation ___CFRunLoopRun
    Frameworks/CoreFoundation.framework/CoreFoundation _CFRunLoopRunSpecific
    PrivateFrameworks/GraphicsServices.framework/GraphicsServices _GSEventRunModal
    Frameworks/UIKit.framework/UIKit _UIApplicationMain
    AcuityApp/main.m:16:12 main
    /usr/lib/system/libdyld.dylib _start
```

**instabug-reactnative:** 8.0.1
**react-native:** 0.56.0
**react:** 16.4.1
**Environment:** 
Prouduction
Devices: iPhone9.2, iPhone10.3, iPhone10.6, iPhone10.5, iPad5.1, iPad7.4, iPad5.2
iOS: 11.2.6, 11.4.0, 11.4.1, 12.0

Seen 198 times so far, happens upon the app initialisation and is caused because there's not enough RAM on the device
On most of those crashes there's only about 50-100mb of ram left at the time of the crash.

I'm not sure why you closed this one https://github.com/Instabug/Instabug-React-Native/issues/183 but this needs to be addressed asap.

I can't give you reproduction steps, as that crashes our app upon launch, and the user doesn't have the chance to do anything.
Is there anything else I can do on my end to help figure this out?

**Comment by Korazy** [comment]
Hey @SudoPlz 

Apologies for the confusion. My colleague Rana is looking into it as we speak.

**Comment by SudoPlz** [comment]
Awesome, thanks @Korazy

**Comment by rdiaa** [comment]
@SudoPlz 
I just want to ask you a question, Is the crash consistent?  does it always crash on launch?

**Comment by SudoPlz** [comment]
It's not consistent, it only happens when the device has no ram left, and I have no idea why that happens, some sort of a memory leak maybe?

**Comment by rdiaa** [comment]
@SudoPlz 
Thank you for providing me with this info, I will investigate and get back to you

**Comment by rdiaa** [comment]
@SudoPlz 
We are still working on this issue and it might take some time, I just wanted to keep you posted with the progress.

**Comment by SudoPlz** [comment]
Thanks for letting me know @rdiaa

**Comment by rdiaa** [comment]
@SudoPlz 
My pleasure :)

**Comment by rdiaa** [comment]
@SudoPlz  I know it's taking too much time but we actually find the root cause and we are fixing it. This should be released within the coming two weeks. I know you have been so patient and I really do appreciate it. I assure to you that I will do my best to get it released the soonest

**Comment by rdiaa** [comment]
@SudoPlz  Please also let me know if its fine for your regarding the release date.

**Comment by SudoPlz** [comment]
Thank you so much @rdiaa , to be honest I didn't really had hopes that you'd resolve this, so I'm really happy to hear you guys managed to figure this out. Memory issues are super weird.

Let me know whenever you have a new public release, I'll be happy to upgrade our Instabug SDK, and make a new release of our app.

Thank you so much

**Comment by rdiaa** [comment]
@SudoPlz Perfect, the new version that includes the fix will be released within two weeks. I will update you once it's released.

It's always my pleasure to help you!

**Comment by bz123** [comment]
@rdiaa we are having the same issue, whats the ETA on the fix? Thanks.

**Comment by SudoPlz** [comment]
@rdiaa it's been 21 days, any news on that?

Thank you

**Comment by HeshamMegid** [comment]
We just fixed this in the latest [iOS release](https://github.com/Instabug/Instabug-iOS/releases/tag/8.0.15). We're releasing the update to React Native later today.

**Comment by HeshamMegid** [comment]
We just released v8.0.16 that includes this fix.
